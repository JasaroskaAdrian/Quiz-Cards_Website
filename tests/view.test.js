

const { view } = require("../src/view");

describe('view function', () => {
    const dispatch = jest.fn(); // https://chatgpt.com - jest.fn
    const model = {
        question: "",
        answer: "",
        items: [],
    };

    test('renders initial view', () => {
        const viewTest = view(dispatch, model);
        expect(viewTest).toMatchSnapshot();
    });

    test('renders view with items', () => {
        const modelWithItems = {
            ...model,
            items: [
                { question: 'Q1', answer: 'A1', showAnswer: false, rating: 0 },
                { question: 'Q2', answer: 'A2', showAnswer: true, rating: 1 },
            ],
        };
        const viewTest = view(dispatch, modelWithItems);
        expect(viewTest).toMatchSnapshot(); // https://chatgpt.com - .ToMatchSnapshot
    });

    test('dispatches SET_QUESTION on input change', () => {
        const viewTest = view(dispatch, model);
        const input = viewTest.children[0].children[0];
        input.properties.oninput({ target: { value: 'New Question' } });
        expect(dispatch).toHaveBeenCalledWith(MSGS.SET_QUESTION, 'New Question'); // https://chatgpt.com - ToHaveBeenCalledWith
    });

    test('dispatches SET_ANSWER on input change', () => {
        const viewTest = view(dispatch, model);
        const input = viewTest.children[0].children[1];
        input.properties.oninput({ target: { value: 'New Answer' } });
        expect(dispatch).toHaveBeenCalledWith(MSGS.SET_ANSWER, 'New Answer'); // https://chatgpt.com - ToHaveBeenCalledWith
    });

    test('dispatches ADD_CARD on button click', () => {
        const viewTest = view(dispatch, model);
        const button = viewTest.children[0].children[2];
        button.properties.onclick();
        expect(dispatch).toHaveBeenCalledWith(MSGS.ADD_CARD); // https://chatgpt.com - ToHaveBeenCalledWith
    });
});