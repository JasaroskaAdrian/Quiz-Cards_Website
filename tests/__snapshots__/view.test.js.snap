// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view function renders initial view 1`] = `
VirtualNode {
  "children": [
    VirtualNode {
      "children": [
        VirtualNode {
          "children": [],
          "count": 0,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "w-full flex-grow p-4 bg-gray-800 text-white border-none rounded-lg shadow-inner focus:border-yellow-400 focus:outline-none focus:shadow-[inset_0px_0px_10px_rgba(255,102,0,0.5),inset_0px_0px_10px_rgba(255,212,59,0.5),0px_0px_100px_rgba(255,212,59,0.5),0px_0px_100px_rgba(255,102,0,0.5)] transition-all duration-200 ease-in-out",
            "oninput": [Function],
            "placeholder": "Question",
            "type": "text",
            "value": SoftSetHook {
              "value": "",
            },
          },
          "tagName": "INPUT",
        },
        VirtualNode {
          "children": [],
          "count": 0,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "w-full flex-grow p-4 bg-gray-800 text-white border-none rounded-lg shadow-inner focus:border-yellow-400 focus:outline-none focus:shadow-[inset_0px_0px_10px_rgba(255,102,0,0.5),inset_0px_0px_10px_rgba(255,212,59,0.5),0px_0px_100px_rgba(255,212,59,0.5),0px_0px_100px_rgba(255,102,0,0.5)] transition-all duration-200 ease-in-out",
            "oninput": [Function],
            "placeholder": "Answer",
            "type": "text",
            "value": SoftSetHook {
              "value": "",
            },
          },
          "tagName": "INPUT",
        },
        VirtualNode {
          "children": [
            VirtualText {
              "text": "Add Flashcard",
            },
          ],
          "count": 1,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "cursor-pointer transition-all bg-blue-500 text-white px-6 py-2 rounded-lg border-blue-600 border-b-[4px] hover:brightness-110 hover:-translate-y-[1px] hover:border-b-[6px] active:border-b-[2px] active:brightness-90 active:translate-y-[2px]",
            "onclick": [Function],
          },
          "tagName": "BUTTON",
        },
      ],
      "count": 4,
      "descendantHooks": false,
      "hasThunks": false,
      "hasWidgets": false,
      "hooks": undefined,
      "key": undefined,
      "namespace": null,
      "properties": {
        "className": "flex flex-col sm:flex-row bg-gradient-to-b from-gray-800 to-black rounded-xl shadow-[10px_10px_20px_#0e1013,-10px_-10px_40px_#383e4b] p-1 gap-1",
      },
      "tagName": "DIV",
    },
  ],
  "count": 5,
  "descendantHooks": false,
  "hasThunks": false,
  "hasWidgets": false,
  "hooks": undefined,
  "key": undefined,
  "namespace": null,
  "properties": {
    "className": "flex flex-col gap-4 items-center p-4",
  },
  "tagName": "DIV",
}
`;

exports[`view function renders view with items 1`] = `
VirtualNode {
  "children": [
    VirtualNode {
      "children": [
        VirtualNode {
          "children": [],
          "count": 0,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "w-full flex-grow p-4 bg-gray-800 text-white border-none rounded-lg shadow-inner focus:border-yellow-400 focus:outline-none focus:shadow-[inset_0px_0px_10px_rgba(255,102,0,0.5),inset_0px_0px_10px_rgba(255,212,59,0.5),0px_0px_100px_rgba(255,212,59,0.5),0px_0px_100px_rgba(255,102,0,0.5)] transition-all duration-200 ease-in-out",
            "oninput": [Function],
            "placeholder": "Question",
            "type": "text",
            "value": SoftSetHook {
              "value": "",
            },
          },
          "tagName": "INPUT",
        },
        VirtualNode {
          "children": [],
          "count": 0,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "w-full flex-grow p-4 bg-gray-800 text-white border-none rounded-lg shadow-inner focus:border-yellow-400 focus:outline-none focus:shadow-[inset_0px_0px_10px_rgba(255,102,0,0.5),inset_0px_0px_10px_rgba(255,212,59,0.5),0px_0px_100px_rgba(255,212,59,0.5),0px_0px_100px_rgba(255,102,0,0.5)] transition-all duration-200 ease-in-out",
            "oninput": [Function],
            "placeholder": "Answer",
            "type": "text",
            "value": SoftSetHook {
              "value": "",
            },
          },
          "tagName": "INPUT",
        },
        VirtualNode {
          "children": [
            VirtualText {
              "text": "Add Flashcard",
            },
          ],
          "count": 1,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "cursor-pointer transition-all bg-blue-500 text-white px-6 py-2 rounded-lg border-blue-600 border-b-[4px] hover:brightness-110 hover:-translate-y-[1px] hover:border-b-[6px] active:border-b-[2px] active:brightness-90 active:translate-y-[2px]",
            "onclick": [Function],
          },
          "tagName": "BUTTON",
        },
      ],
      "count": 4,
      "descendantHooks": false,
      "hasThunks": false,
      "hasWidgets": false,
      "hooks": undefined,
      "key": undefined,
      "namespace": null,
      "properties": {
        "className": "flex flex-col sm:flex-row bg-gradient-to-b from-gray-800 to-black rounded-xl shadow-[10px_10px_20px_#0e1013,-10px_-10px_40px_#383e4b] p-1 gap-1",
      },
      "tagName": "DIV",
    },
    VirtualNode {
      "children": [
        VirtualNode {
          "children": [
            VirtualText {
              "text": "Delete",
            },
          ],
          "count": 1,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "absolute top-2 right-2 cursor-pointer transition-all bg-red-500 text-white px-6 py-2 rounded-lg border-red-600 border-b-[4px] hover:brightness-110 hover:-translate-y-[1px] hover:border-b-[6px] active:border-b-[2px] active:brightness-90 active:translate-y-[2px]",
            "onclick": [Function],
          },
          "tagName": "BUTTON",
        },
        VirtualNode {
          "children": [
            VirtualText {
              "text": "Question",
            },
          ],
          "count": 1,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "font-bold mb-2 text-white text-4x1",
          },
          "tagName": "H3",
        },
        VirtualNode {
          "children": [
            VirtualText {
              "text": "Q1",
            },
          ],
          "count": 1,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "text-lg font-semibold mb-4 text-white font-thin",
          },
          "tagName": "P",
        },
        VirtualNode {
          "children": [
            VirtualText {
              "text": "Show Answer",
            },
          ],
          "count": 1,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "bg-yellow-500 text-white px-4 py-2 rounded mb-4",
            "onclick": [Function],
          },
          "tagName": "BUTTON",
        },
      ],
      "count": 8,
      "descendantHooks": false,
      "hasThunks": false,
      "hasWidgets": false,
      "hooks": undefined,
      "key": undefined,
      "namespace": null,
      "properties": {
        "className": "bg-teal-700 p-4 rounded-lg border border-gray-400 w-80 relative shadow-lg",
      },
      "tagName": "DIV",
    },
    VirtualNode {
      "children": [
        VirtualNode {
          "children": [
            VirtualText {
              "text": "Delete",
            },
          ],
          "count": 1,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "absolute top-2 right-2 cursor-pointer transition-all bg-red-500 text-white px-6 py-2 rounded-lg border-red-600 border-b-[4px] hover:brightness-110 hover:-translate-y-[1px] hover:border-b-[6px] active:border-b-[2px] active:brightness-90 active:translate-y-[2px]",
            "onclick": [Function],
          },
          "tagName": "BUTTON",
        },
        VirtualNode {
          "children": [
            VirtualText {
              "text": "Question",
            },
          ],
          "count": 1,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "font-bold mb-2 text-white text-4x1",
          },
          "tagName": "H3",
        },
        VirtualNode {
          "children": [
            VirtualText {
              "text": "Q2",
            },
          ],
          "count": 1,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "text-lg font-semibold mb-4 text-white font-thin",
          },
          "tagName": "P",
        },
        VirtualNode {
          "children": [
            VirtualText {
              "text": "Show Answer",
            },
          ],
          "count": 1,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "bg-yellow-500 text-white px-4 py-2 rounded mb-4",
            "onclick": [Function],
          },
          "tagName": "BUTTON",
        },
        VirtualNode {
          "children": [
            VirtualText {
              "text": "Answer",
            },
          ],
          "count": 1,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "font-bold mb-2 text-white text-4x1",
          },
          "tagName": "H4",
        },
        VirtualNode {
          "children": [
            VirtualText {
              "text": "A2",
            },
          ],
          "count": 1,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "mb-4 text-white font-thin",
          },
          "tagName": "P",
        },
        VirtualNode {
          "children": [
            VirtualNode {
              "children": [
                VirtualText {
                  "text": "Bad",
                },
              ],
              "count": 1,
              "descendantHooks": false,
              "hasThunks": false,
              "hasWidgets": false,
              "hooks": undefined,
              "key": undefined,
              "namespace": null,
              "properties": {
                "className": "bg-red-500 text-white px-4 py-2 rounded",
                "onclick": [Function],
              },
              "tagName": "BUTTON",
            },
            VirtualNode {
              "children": [
                VirtualText {
                  "text": "Good",
                },
              ],
              "count": 1,
              "descendantHooks": false,
              "hasThunks": false,
              "hasWidgets": false,
              "hooks": undefined,
              "key": undefined,
              "namespace": null,
              "properties": {
                "className": "bg-blue-500 text-white px-4 py-2 rounded",
                "onclick": [Function],
              },
              "tagName": "BUTTON",
            },
            VirtualNode {
              "children": [
                VirtualText {
                  "text": "Great",
                },
              ],
              "count": 1,
              "descendantHooks": false,
              "hasThunks": false,
              "hasWidgets": false,
              "hooks": undefined,
              "key": undefined,
              "namespace": null,
              "properties": {
                "className": "bg-green-500 text-white px-4 py-2 rounded",
                "onclick": [Function],
              },
              "tagName": "BUTTON",
            },
          ],
          "count": 6,
          "descendantHooks": false,
          "hasThunks": false,
          "hasWidgets": false,
          "hooks": undefined,
          "key": undefined,
          "namespace": null,
          "properties": {
            "className": "flex justify-around mt-4",
          },
          "tagName": "DIV",
        },
      ],
      "count": 19,
      "descendantHooks": false,
      "hasThunks": false,
      "hasWidgets": false,
      "hooks": undefined,
      "key": undefined,
      "namespace": null,
      "properties": {
        "className": "bg-teal-700 p-4 rounded-lg border border-gray-400 w-80 relative shadow-lg",
      },
      "tagName": "DIV",
    },
  ],
  "count": 34,
  "descendantHooks": false,
  "hasThunks": false,
  "hasWidgets": false,
  "hooks": undefined,
  "key": undefined,
  "namespace": null,
  "properties": {
    "className": "flex flex-col gap-4 items-center p-4",
  },
  "tagName": "DIV",
}
`;
